valet

```diff
@@ -81,16 +75,36 @@ then
     ARCH=$(uname -m)
 
     if [[ $ARCH == 'arm64' ]]; then
-        sudo -u "$USER" "$DIR/bin/ngrok-arm" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
+        "$DIR/bin/ngrok-arm" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
     else
-        sudo -u "$USER" "$DIR/bin/ngrok" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
+        "$DIR/bin/ngrok" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
```

Comment:
Could you explain to me what the change to these lines, and also moving lines 23:28, are for? Sorry, i don't full understand either, and I figured it'll be faster for you to share than for me to dig into it on my own. Thanks!

Reply:
```bash
if [[ "$EUID" -ne 0 ]]
then
    sudo USER="$USER" --preserve-env "$SOURCE" "$@"
    exit
fi
```

Occurred to me, we just need these lines for this particular PHP call `php "$DIR/cli/valet.php" "$@"`. Other commands from this file don't require a root user. So, We moved this chunk of code from the top of this file to the bottom.

Also, we definetly don't want to proxy `php` and `composer` commands with the root user. So moving that line to the bottom kinda made sense to me. 

Now that we moved that line to the bottom. `valet share`, `valet php` and `valet composer` commands should not come from `root` user. So this `sudo -u "$USER"` shouldn't be necessary anymore. 

--

btw, please do let me know, if you think we should be explicit about the `user` we are using to run all these commands, I'll undo this and update my PR accordingly.

Reply:
No, that makes a ton of sense--I just have never *fully* followed which user we need in which context, so it's helpful to have your explanation. :) Thank you!